app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java

```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             mapView = binding.mapView;
             mapView.onCreate(savedInstanceState);
             mapView.getMapAsync(this);
-
             initializeAvailableColors();
 
             binding.walkingRoutesFilterScrollView.setAlpha(TRANSPARENCY);
         }
 
+        flSharePetWalkRouteButton = binding.flSharePetWalkRouteButton;
+        setUpSharePetWalkRouteListener();
         return binding.getRoot();
     }
 
+    /**
+     * Set the share app button listener.
+     */
+    private void setUpSharePetWalkRouteListener() {
+        flSharePetWalkRouteButton.setOnClickListener(v -> {
+            GoogleMap.SnapshotReadyCallback callback = new GoogleMap.SnapshotReadyCallback() {
+            @Override
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['method def modifier' has incorrect indentation level 12, expected level should be one of the following: 16, 20.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)